shelley/chain-and-ledger/executable-spec/test/Test/Shelley/Spec/Ledger/Address.hs

```diff
@@ -81,19 +84,69 @@ goldenTests =
         "addrEnterpriseS"
         putAddr
         (Addr scriptHash StakeRefNull)
-        "7205060708"
+        "7205060708",
+
+      -- serialisation golden tests
+      goldenSerialisation
+        "addrEnterpriseK for network id = 0"
+        (Addr (keyBlake2b224 paymentKey) StakeRefNull)
+        "62",
+      goldenSerialisation
+        "addrBaseKK for network id = 0"
+        (Addr (keyBlake2b224 paymentKey) (StakeRefBase (keyBlake2b224 stakeKey)))
+        "02",
+      goldenSerialisation
+        "addrPtrK for network id = 0"
+        (Addr (keyBlake2b224 paymentKey) (StakeRefPtr ptr))
+        "4281000203"
+
+       -- the same as above for other network ids is also needed, for example, for 3, 6
     ]
 
 golden :: String -> (a -> B.Put) -> a -> LBS.ByteString -> TestTree
 golden name put value expected =
   T.testCase name $
     T.assertEqual name expected (LB16.encode . B.runPut . put $ value)
 
+goldenSerialisation :: String -> C.Addr -> BS.ByteString -> TestTree
+goldenSerialisation name value expected =
+  T.testCase name $
+    T.assertEqual name expected (B16.encode . serialiseAddr $ value)
+
 keyHash :: C.Credential kh
 keyHash =
   KeyHashObj . KeyHash . UnsafeHash . fst $
     B16.decode "01020304"
 
+paymentKey :: BS.ByteString
+paymentKey = B16.encode "1a2a3a4a5a6a7a8a"
+
+stakeKey :: BS.ByteString
+stakeKey = B16.encode "1c2c3c4c5c6c7c8c"
+
+-- 32-byte verification key is expected, vk, ie., public key without chain code.
+-- The verification key undergoes Blake2b_224 hashing
+-- and should be 28-byte in the aftermath
+keyBlake2b224 :: BS.ByteString ->  C.Credential kh
+keyBlake2b224 vk =
+  KeyHashObj . KeyHash . UnsafeHash . fst $
+    B16.decode hk
```

Comment:
I think you don't want `B16.decode` here.

Reply:
indeed -> corrected here https://github.com/input-output-hk/cardano-ledger-specs/pull/1443/commits/8c7a0d69f361f4195edd7ed89203813d7aac3d77